packages/delegation-process/test/createConsumerDelegation.test.ts

```diff
@@ -0,0 +1,575 @@
+          availabilityTimestamp: currentExecutionTime,
+        },
+      ],
+    };
+    const eservice = getMockEService(generateId<EServiceId>(), delegatorId);
+
+    await addOneTenant(delegator);
+    await addOneTenant(delegate);
+    await addOneEservice(eservice);
+
+    const actualDelegation =
+      await delegationConsumerService.createConsumerDelegation(
+        {
+          delegateId: delegate.id,
+          eserviceId: eservice.id,
+        },
+        {
+          authData,
+          logger: genericLogger,
+          correlationId: generateId(),
+          serviceName: "DelegationServiceTest",
+        }
+      );
+
+    const expectedDelegation: Delegation = {
+      id: actualDelegation.id,
+      delegatorId,
+      delegateId: delegate.id,
+      eserviceId: eservice.id,
+      kind: delegationKind.delegatedConsumer,
+      state: delegationState.waitingForApproval,
+      createdAt: currentExecutionTime,
+      submittedAt: currentExecutionTime,
+      stamps: {
+        submission: {
+          who: delegatorId,
+          when: currentExecutionTime,
+        },
+      },
+    };
+
+    await expectedDelegationCreation(actualDelegation, expectedDelegation);
+    vi.useRealTimers();
+  });
+
+  it.each(inactiveDelegationStates)(
+    "should create a delegation the same delegation exists and is in state %s",
+    async (inactiveDelegationState) => {
+      const currentExecutionTime = new Date();
+      vi.useFakeTimers();
+      vi.setSystemTime(currentExecutionTime);
+
+      const delegatorId = generateId<TenantId>();
+      const authData = getRandomAuthData(delegatorId);
+      const delegator = {
+        ...getMockTenant(delegatorId),
+        externalId: {
+          origin: "IPA",
+          value: "test",
+        },
+      };
+
+      const delegate = {
+        ...getMockTenant(),
+        features: [
+          {
+            type: "DelegatedConsumer" as const,
+            availabilityTimestamp: currentExecutionTime,
+          },
+        ],
+      };
+      const eservice = getMockEService(generateId<EServiceId>(), delegatorId);
+
+      const existentDelegation = {
+        ...getMockDelegation({
+          kind: delegationKind.delegatedConsumer,
+          id: generateId<DelegationId>(),
+          delegatorId,
+          delegateId: delegate.id,
+          eserviceId: eservice.id,
+        }),
+        state: inactiveDelegationState,
+      };
+
+      await addOneTenant(delegator);
+      await addOneTenant(delegate);
+      await addOneEservice(eservice);
+      await addOneDelegation(existentDelegation);
+
+      const actualDelegation =
+        await delegationConsumerService.createConsumerDelegation(
+          {
+            delegateId: delegate.id,
+            eserviceId: eservice.id,
+          },
+          {
+            authData,
+            logger: genericLogger,
+            correlationId: generateId(),
+            serviceName: "DelegationServiceTest",
+          }
+        );
+
```

Comment:
I would also add a check that the new `delegationId` is different from the old one